packages/kbn-kibana-manifest-parser/src/kibana_manifest.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+export type KibanaPackageType =
+  | 'plugin-browser'
+  | 'plugin-server'
+  | 'shared-browser'
+  | 'shared-server'
+  | 'shared-common'
+  | 'functional-tests'
+  | 'test-helper';
+
+interface PackageManifestBaseFields {
+  type: KibanaPackageType;
+  id: string;
+  owner: string;
+  typeDeps: string[];
+  runtimeDeps: string[];
+  tsconfigType?: string;
```

Comment:
do we need this field? Can't it be inferred automatically from the package type?

Reply:
This is necessary for teams to override and customize their tsconfig files if they want to, for storybook stuff for instance. We'll be able to set the default based on package type, and maybe we can defer this customization but it's something that @clintandrewhall was looking to accomplish pretty early.

Reply:
I see. Should it be called ` tsconfigOverride` or something like this in that case?

Reply:
Yeah, maybe the name is the problem... `baseTsconfig` maybe?

Reply:
Actually, this is going to require a `shared-tsconfig` package type or something so that the primary export is a `tsconfig.json` file, so maybe we should just remove this for now.